packages/web/src/components/track/VisibilityLabel.tsx

```diff
@@ -27,34 +22,22 @@ export type VisibilityLabelProps = {
 
 export const VisibilityLabel = (props: VisibilityLabelProps) => {
   const { releaseDate, isUnlisted, isScheduledRelease } = props
-  const { color } = useTheme()
-
-  if (
-    !releaseDate ||
-    !isUnlisted ||
-    !isScheduledRelease ||
-    dayjs(releaseDate).isBefore(dayjs())
-  ) {
-    return null
+
+  if (releaseDate && isUnlisted && isScheduledRelease) {
+    return (
+      <LineupTileLabel icon={IconCalendarMonth} color='accent'>
+        {messages.releases(releaseDate)}
+      </LineupTileLabel>
+    )
   }
 
-  if (isUnlisted && !isScheduledRelease) {
+  if (isUnlisted) {
     return (
-      <Flex alignItems='center' gap='xs'>
-        <IconVisibilityHidden size='s' color='subdued' />
-        <Text variant='body' size='xs' color='subdued'>
-          {messages.hidden}
-        </Text>
-      </Flex>
+      <LineupTileLabel icon={IconVisibilityHidden} color='subdued'>
+        {messages.hidden}
+      </LineupTileLabel>
     )
   }
 
-  return (
-    <Flex alignItems='center' gap='xs'>
-      <IconCalendarMonth size='s' fill={color.icon.accent} />
-      <Text variant='body' size='xs' color='accent'>
-        {messages.releases(releaseDate)}
-      </Text>
-    </Flex>
-  )
+  return null
```

Comment:
it feels weird to have the return null at the end, but the if conditions are a lot cleaner so I agree with the changes

Reply:
yeah kinda a catch all situation, but yeah odd agreed